File: src/main/java/miretz/ycloud/models/Document.java
package miretz.ycloud.models;

import java.util.Map;

public class Document {

	public static final String TYPE_FILE = "file";
	public static final String TYPE_FOLDER = "folder";

	private final String contentId;
	private final String fileName;
	private final String parentId;
	private final Map<String, String> metadata;
	private final String type;

	public Document(String contentId, String fileName, String parentId, Map<String, String> metadata, String type) {
		super();
		this.contentId = contentId;
		this.fileName = fileName;
		this.parentId = parentId;
		this.metadata = metadata;
		this.type = type;
	}

	public String getContentId() {
		return contentId;
	}

	public String getFileName() {
		return fileName;
	}

	public String getParentId() {
		return parentId;
	}

	public Map<String, String> getMetadata() {
		return metadata;
	}

	public String getType() {
		return type;
	}

}

File: src/main/java/miretz/ycloud/models/Document.kt
package miretz.ycloud.models

import java.util.*

class Document(val contentId: String, val fileName: String, val parentId: String, val metadata: Map<String, String>, val type: String, val retentionDate: Date?) {
    companion object {
        val TYPE_FILE: String = "file"
        val TYPE_FOLDER: String = "folder"
    }

}
